components/feature/session/src/main/java/mozilla/components/feature/session/SessionIntentProcessor.kt

```diff
@@ -10,25 +10,33 @@ import mozilla.components.browser.session.Session
 import mozilla.components.browser.session.Session.Source
 import mozilla.components.browser.session.SessionManager
 import mozilla.components.browser.session.tab.CustomTabConfig
+import mozilla.components.support.ktx.kotlin.isUrl
 import mozilla.components.support.utils.SafeIntent
 
 typealias IntentHandler = (Intent) -> Boolean
 
+typealias TextSearchHandler = (String, Session) -> Unit
+
 /**
  * Processor for intents which should trigger session-related actions.
  *
  * @param openNewTab Whether a processed Intent should open a new tab or open URLs in the currently
  *                   selected tab.
+ *
+ * @param textSearchHandler This lambda expression will be invoked when a text search should be performed,
+ *                           the consumer must provide a behavior to handle the search, by default it will do nothing.
```

Comment:
I think we can be more concrete here i.e. when exactly is this invoked. So instead of "This lambda expression..." (which can be seen based on the type) I would write "textSearchHandler Will be invoked for ACTION_SEND intents when the provided dataString is not a URL..."